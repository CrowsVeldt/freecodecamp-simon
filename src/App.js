import React, { Component } from 'react'
import styled from 'styled-components'
import ColorButton from './ColorButton'

class App extends Component {
  render () {
    return (
      <Game className='App'>
        <CenterPanel>
          <h1>Simon</h1>
        </CenterPanel>
        <ColorButton green source='https://s3.amazonaws.com/freecodecamp/simonSound1.mp3' />
        <ColorButton red source='https://s3.amazonaws.com/freecodecamp/simonSound2.mp3' />
        <ColorButton blue source='https://s3.amazonaws.com/freecodecamp/simonSound3.mp3' />
        <ColorButton yellow source='https://s3.amazonaws.com/freecodecamp/simonSound4.mp3' />
      </Game>
    )
  }
}

const Game = styled.div`
border: 1px solid black;
border-radius: 50%;
height: 700px;
width: 700px;
box-shadow: 5px 5px 2.5px 0px rgba(0,0,0,0.5);
display: flex;
justify-content: center;
align-items: center;
position: relative;
z-index: 0;
`
const CenterPanel = styled.div`
border: 1px solid black;
border-radius: 50%;
height: 300px;
width: 300px;
display: flex;
justify-content: center;
align-items: center;

background-color: black;
color: white;

z-index: 2;
`

export default App
