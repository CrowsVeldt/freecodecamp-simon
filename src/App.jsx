import ColorButton from './ColorButton'
import React, { Component } from 'react'
import styled from 'styled-components'

class App extends Component {
  constructor (props) {
    super(props)

    this.state = {
      sequence: [],
      isPlaying: false
    }
  }

  startGame = () =>  {
    const currentSequence = this.state.sequence
    currentSequence.push(generateNumber())
    this.setState({
      sequence: currentSequence,
      isPlaying: true
    })
  }

  render () {
    return (
      <Game className='game'>

        <CenterPanel className='centerPanel'>
          <h1>
            Simon
          </h1>
          <ButtonContainer className='buttonContainer'>
            <PlayButton onClick={ this.startGame }>
              Play
            </PlayButton>
          </ButtonContainer>
        </CenterPanel>

        <ColorButton
          // ref={(greenButton) => {this.greenButtonRef = greenButton}}
          green
          source={'https://s3.amazonaws.com/freecodecamp/simonSound1.mp3'}
        />

        <ColorButton
          red
          source={'https://s3.amazonaws.com/freecodecamp/simonSound2.mp3'}

        />

        <ColorButton
          blue
          source={'https://s3.amazonaws.com/freecodecamp/simonSound3.mp3'}
        />

        <ColorButton
          yellow
          source={'https://s3.amazonaws.com/freecodecamp/simonSound4.mp3'}
        />
      </Game>
    )
  }
}

const Game = styled.div`
border: 1px solid black;
border-radius: 50%;
height: 700px;
width: 700px;
box-shadow: 5px 5px 2.5px 0px rgba(0,0,0,0.5);
display: flex;
justify-content: center;
align-items: center;
position: relative;
z-index: 0;
`
const CenterPanel = styled.div`
border: 1px solid black;
border-radius: 50%;
height: 300px;
width: 300px;
display: flex;
flex-direction: column;
justify-content: center;
align-items: center;

background-color: black;
color: white;

z-index: 2;
`

const ButtonContainer = styled.div`
background-color: tan;
`

const PlayButton = styled.button`
border-radius: 100%;
height: 60px;
width: 60px;
background-color: green;
color: white;
`

export default App

export function generateNumber () {
  const min = 0
  const max = 3
  return Math.floor(Math.random() * (max - min + 1)) + min
}
