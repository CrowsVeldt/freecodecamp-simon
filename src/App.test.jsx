import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import App, {generateNumber} from './App'

// Organize the code into describe blocks

describe('The App component', () => {
  
  test('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<App />, div)
  })

  test('renders correctly', () => {
    const tree = renderer.create(<App />)
    expect(tree).toMatchSnapshot()
  })
})


describe('generateNumber', () => {

  const arrayOfRandoms = []

  for (let i = 0; i < 100; i++) {
    arrayOfRandoms.push(generateNumber())
  }

  test('always returns a number', () => {
    arrayOfRandoms.map(item => {
      expect(typeof item).toBe('number')
    })
  })

  test('always returns a whole number', () => {
    arrayOfRandoms.map(item => {
      expect(item % 1).toBe(0)
    })
  })

  test('does not return any number greater than 3', () => {
    arrayOfRandoms.map(item => {
      expect(item).toBeLessThan(4)
    })
  })

  test('does not return any number less than 0', () => {
    arrayOfRandoms.map(item => {
      expect(item).toBeGreaterThanOrEqual(0)
    })
  })
})
