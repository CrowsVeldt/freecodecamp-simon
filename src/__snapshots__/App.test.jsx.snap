// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="game sc-bwzfXH jEGHCr"
>
  <div
    className="centerPanel sc-htpNat SFyyG"
  >
    <h1>
      Simon
    </h1>
    <div
      className="buttonContainer sc-bxivhb cfzwfq"
    >
      <button
        className="sc-ifAKCX bRRogu"
      >
        Play
      </button>
    </div>
  </div>
  <button
    className="sc-bdVaJa kdmWDA"
    onClick={[Function]}
  >
    <audio>
      <source
        src="https://s3.amazonaws.com/freecodecamp/simonSound1.mp3"
      />
    </audio>
  </button>
  <button
    className="sc-bdVaJa kIosMh"
    onClick={[Function]}
  >
    <audio>
      <source
        src="https://s3.amazonaws.com/freecodecamp/simonSound2.mp3"
      />
    </audio>
  </button>
  <button
    className="sc-bdVaJa bWISYV"
    onClick={[Function]}
  >
    <audio>
      <source
        src="https://s3.amazonaws.com/freecodecamp/simonSound3.mp3"
      />
    </audio>
  </button>
  <button
    className="sc-bdVaJa jLtlvl"
    onClick={[Function]}
  >
    <audio>
      <source
        src="https://s3.amazonaws.com/freecodecamp/simonSound4.mp3"
      />
    </audio>
  </button>
</div>
`;
